modules/admin_manual/pages/enterprise/external_storage/enterprise_only_auth.adoc

```diff
@@ -1,45 +1,205 @@
 
-Use the dropdown selector to choose the authentication backend when you
-create a new external mount.
+The following table shows which authentication option allows sharing:
+ 
+[width="70%",cols="50%,^20%",",options="header"]
+|===
+| Authentication Option
+| Sharing Allowed
 
-image:enterprise/external_storage/enterprise-only-auth/auth_backends-1.png[Authentication dropdown selector.]
+| Username and password
+| x
+
+| Log-in credentials, save in session
+|
+
+| Log-in credentials, save in database
+| x
+
+| User entered, store in database
+| x
+
+| Credentials hardcoded in config file
+| x
+
+| Global credentials
+| x
+
+| RSA public key
+| x
+|===
+
+=== Notes for Credentials Hardcoded in Config File
+
+* The username taken will be the same as the `user id` of the ownCloud session. For example, ownCloud user "Alice" with password "mysecret" (or even without password) will use "Alice" as username and the password from the config.php file to access the storage. Note that for LDAP, the `user id` is usually like _1223-cbdf-cacc-1234...,_ unless the configuration in the user_ldap app is changed.
+
+* The password will be fetched from a key inside the config.php file. For details see the section below. Even if the account doesn't have a password like oAuth or OpenIDConnect etc, the connection with the storage will use the password from  config.php. *Note that the password will be the same for any user accessing the particular mount!*
+
+==== Defining key/value pairs in config.php
+
+Key/value pairs in config.php must have the following structure:
+
+.Example having a single array
+[source,php]
+----
+'customApp.config' => [
+    'password' => 'the_password',
+]
+----
+
+.Example having nested arrays
+[source,php]
+----
+'customApp.config' => [
+  'server1' => [
+    'password' => 'the_first_password',
+  ],
+  'server2' => [
+    'password' => 'the_second_password',
+  ],
+  ....
+]
+----
+
+customApp.config::
+The naming of this key must be a string that is valid as an array key in a PHP array like the above `customApp.config` or as another example `my.config.key`, but not any reserved ownCloud key.
+
+password::
+Each array element **must** have exactly one key named `password` present and a corresponding value, even if empty.
+
+Nested arrays::
+You can use nested arrays as described in the example above. When using nested arrays, you can use as many sub-keys according your needs where the naming of the sub-key must be a string that is valid as an array key in a PHP array.
++
+Note that sub-keys, except the password key are not used as elements for the mount point. Sub-keys are only present to optimize grouping and accessing the passwords.
++
+Nested arrays are beneficial when having more than one server (host) with a password, but keeping them together in one master key. To access a particular sub-key in the mount definition, use the following scheme:
++
+[source,text]
+----
+<key>#<level_1>#<level_2>...
+----
```

Comment:
Hard to understand for me. What is a subkey? What is key? (probably a servername, from the exampe below) Where do levels appear?
Where exactly is this syntax used? In the entry form of the admin sharing dialog?

Reply:
fixed, text added